sympy/printing/latex.py

```diff
@@ -446,9 +446,17 @@ def convert(expr):
             snumer = convert(numer)
             sdenom = convert(denom)
             ldenom = len(sdenom.split())
-            ratio = self._settings['long_frac_ratio']
+            if self._settings['long_frac_ratio'] == None:
+                exprfind = [x for x in expr.args if \
+                    str(x).startswith('1/')]
```

Comment:
This should be tested directly. You shouldn't use the str form of an expression to test that it has some form.

Reply:
The ```startswith``` method or regex functions only support strings. So how can I check if a 'rational' object starts with something without converting into a string?

Reply:
If you really mean an object of class Rational, there is property "p" that holds the numerator. But probably there are other classes involved. Maybe you want to count the number of terms in the numerator or denominator. The method `as_numer_denom` may be helpful then. 

But I don't understand why you are doing this. Was the idea to prevent splitting when ratio is None? Then the place to do that would be a bit below, where the condition `len(snumer.split()) > ratio*ldenom` would be replaced by 
`ratio is not None and len(snumer.split()) > ratio*ldenom`. No need to come up with some value for "ratio" when it's None.

Reply:
@normalhuman I modified the condition but there seems to be several failed test cases. Can you please tell me what is wrong with my code? I just wanted to change the value of "ratio" according to the number of terms in the numerator or denominator and my code seems to do it just fine.

Reply:
What's wrong was already stated above: "You shouldn't use the str form of an expression to test that it has some form"

Reply:
@normalhuman  Yes I understand that but as I have already stated that the ```startswith``` method or regex functions only support strings so how can I check if a 'rational' object starts with something without converting into a string? I need it to check because I need the correct number of terms in numerator and denominator and without it the output might be more than expected if the expression contains constant terms or so.

Reply:
The str test is unacceptable. So you either find an acceptable solution, or find something else to do.